### ISSUE TITLE
Clipboard not syncing across devices

### ISSUE THREAD

**Issue by alduraibi** [andyholmes/valent#616]
### Current Behavior

The clipboard does not sync between my devices. I currently have Valent and its extension installed on 3 devices: 2 on Arch and one on postmarketOS. I also have a forth device running KDE Connect on Android. Clilboard syncing used to work pretty well when I used GSConnect but since i switched to Valent, it has been the most noticeable issue. The clipboard does not send or recieve anything from other devices. It sometimes works at the first boot before a device goes go sleep but even this appears to rarely happen. 

### Expected Behavior

The clipboard should sync effortlessly between all my devices.

Extra: Would be nice to have the clipboard syncing feature enabled by default when connecting a new device.

### Desktop

GNOME Shell

### Other Desktop

and GNOME-mobile

### Operating System

Arch and postmarketOS

### Installed from

Nightly Flatpak

### Version

1.0.0.alpha

### Devices

KDE Connect (Android), Valent (Desktop Linux), Valent (Phosh)

### Plugins

Clipboard

**Comment by andyholmes** [comment]
If you can get some logs of when it happens, that would be helpful. I've tested this on GNOME Mobile, Phosh and Fedora Linux without problems, so it should work.

Maybe there is something different about Arch I'm not familiar with that's tripping it up.

**Comment by vredesbyyrd** [comment]
Just to chime in as an Arch user - This is not an issue I have run into, yet. Clipboard syncing is working both ways consistently. If that changes I will report back.

**Comment by ghassan0** [comment]
Here are my logs. Seems to be an issue with the Clipboard service not being available on my Arch laptops.

## Arch Laptop 1
Clipboard syncing didn't work at all from boot.

<details>

<summary>Logs:</summary>

```
14:00:46 valent: 14:00:46.7365        valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_connect_cb(): Clipboard service not available.
14:00:46 valent: 14:00:46.7365        valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_connect_cb(): Clipboard service not available.
14:00:31 valent: 14:00:31.9347        valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_connect_cb(): Clipboard service not available.
14:00:31 valent: 14:00:31.8609        valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_connect_cb(): Clipboard service not available.
14:00:31 valent: 14:00:31.7178     valent-lan-channel-service:  WARNING: valent_lan_channel_service_socket_send(): failed to identify: Error sending message: Cannot assign requested address
Jan 19 17:19:20 valent: 17:19:20.2479             valent-ebook-store:    DEBUG: ValentEBookStore (Address Book): failed to connect backend: Timeout was reached
Jan 19 17:19:20 valent: 17:19:20.2459             valent-ebook-store:    DEBUG: ValentEBookStore (Address Book): failed to connect backend: Timeout was reached
Jan 19 17:19:01 gnome-software: Warning: Treating remote fetch error as non-fatal since app/ca.andyholmes.Valent/x86_64/master is already installed: No such ref 'app/ca.andyholmes.Valent/x86_64/master' in remote valent-origin
Jan 19 17:19:01 gnome-software: Warning: Treating remote fetch error as non-fatal since app/ca.andyholmes.Valent/x86_64/master is already installed: No such ref 'app/ca.andyholmes.Valent/x86_64/master' in remote valent-origin
Jan 19 17:19:01 gnome-software: Warning: Treating remote fetch error as non-fatal since runtime/ca.andyholmes.Valent.Locale/x86_64/master is already installed: No such ref 'runtime/ca.andyholmes.Valent.Locale/x86_64/master' in remote valent-origin
Jan 19 17:18:56 gnome-software: Warning: Treating remote fetch error as non-fatal since app/ca.andyholmes.Valent/x86_64/master is already installed: No such ref 'app/ca.andyholmes.Valent/x86_64/master' in remote valent-origin
Jan 19 17:18:56 gnome-software: Warning: Treating remote fetch error as non-fatal since runtime/ca.andyholmes.Valent.Locale/x86_64/master is already installed: No such ref 'runtime/ca.andyholmes.Valent.Locale/x86_64/master' in remote valent-origin
Jan 19 17:18:50 valent: 17:18:50.1924                       GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation portal (GNetworkMonitorPortal) for ‘gio-network-monitor’
Jan 19 17:18:50 valent: 17:18:50.1924                       GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation portal (GNetworkMonitorPortal) for ‘gio-network-monitor’
Jan 19 17:18:49 valent: 17:18:49.9613                       GLib-GIO:    DEBUG: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
Jan 19 17:18:49 valent: 17:18:49.9020                       GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation gnutls (GTlsBackendGnutls) for ‘gio-tls-backend’
Jan 19 17:18:49 valent: 17:18:49.9000                       GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation keyfile (GKeyfileSettingsBackend) for ‘gsettings-backend’
Jan 19 17:18:49 valent: 17:18:49.7849                       GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
Jan 19 17:18:49 valent: 17:18:49.5746                       GLib-GIO:    DEBUG: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
Jan 19 17:18:49 systemd: Started app-flatpak-ca.andyholmes.Valent-1183.scope.

```

</details>

<details>

<summary>Debug info:</summary>

```
{
  "application" : {
    "id" : "ca.andyholmes.Valent",
    "version" : "1.0.0.alpha",
    "commit" : "508b7ca"
  },
  "runtime" : {
    "os" : "GNOME 45 (Flatpak runtime)",
    "desktop" : "GNOME",
    "session" : "wayland",
    "environment" : "flatpak"
  },
  "plugins" : {
    "xdp" : true,
    "telephony" : true,
    "systemvolume" : true,
    "sms" : true,
    "share" : true,
    "sftp" : true,
    "runcommand" : true,
    "pulseaudio" : true,
    "presenter" : true,
    "ping" : true,
    "notification" : true,
    "mpris" : true,
    "mousepad" : true,
    "lock" : true,
    "libvalent-ui" : true,
    "libvalent-device" : true,
    "lan" : true,
    "gtk" : true,
    "gnome" : true,
    "findmyphone" : true,
    "fdo" : true,
    "eds" : true,
    "contacts" : true,
    "connectivity_report" : true,
    "clipboard" : true,
    "battery" : true
  }
}
```

</details>

## Arch Laptop 2
Right after boot, clipboard syncing worked to sync to my Android phone, then quickly stopped syncing anything.

<details>

<summary>Logs:</summary>

```
14:42:38 valent: 14:42:38.1519        valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_connect_cb(): Clipboard service not available.
14:42:38 valent: 14:42:38.1519        valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_connect_cb(): Clipboard service not available.
14:42:11 valent: 14:42:11.6280        valent-clipboard-plugin:  WARNING: valent_clipboard_write_text_cb(): Clipboard service not available.
14:39:54 valent: 14:39:54.5018        valent-clipboard-plugin:    DEBUG: valent_clipboard_read_text_cb(): No such interface “org.gnome.Mutter.RemoteDesktop.Session” on object at path /org/gnome/Mutter/RemoteDesktop/Session/u1
14:39:54 valent: 14:39:54.5018        valent-clipboard-plugin:    DEBUG: valent_clipboard_read_text_cb(): No such interface “org.gnome.Mutter.RemoteDesktop.Session” on object at path /org/gnome/Mutter/RemoteDesktop/Session/u1
14:39:54 valent: 14:39:54.5017        valent-clipboard-plugin:    DEBUG: valent_clipboard_read_text_cb(): No such interface “org.gnome.Mutter.RemoteDesktop.Session” on object at path /org/gnome/Mutter/RemoteDesktop/Session/u1
14:39:04 valent: 14:39:04.9983        valent-clipboard-plugin:    DEBUG: valent_clipboard_read_text_cb(): Tried to read in parallel
14:39:04 valent: 14:39:04.9982        valent-clipboard-plugin:    DEBUG: valent_clipboard_read_text_cb(): Tried to read in parallel
14:34:35 valent: 14:34:35.5215             valent-ebook-store:    DEBUG: ValentEBookStore (Address Book): failed to connect backend: Timeout was reached
14:34:35 valent: 14:34:35.5209             valent-ebook-store:    DEBUG: ValentEBookStore (Address Book): failed to connect backend: Timeout was reached
14:34:08 valent: 14:34:08.3473        valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_connect_cb(): text not available
14:34:08 valent: 14:34:08.0049     valent-lan-channel-service:  WARNING: valent_lan_channel_service_socket_send(): failed to identify: Error sending message: Cannot assign requested address
14:34:07 valent: 14:34:07.7426        valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_connect_cb(): text not available
14:34:07 valent: 14:34:07.5025        valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_connect_cb(): text not available
14:34:07 valent: 14:34:07.4840        valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_connect_cb(): text not available
14:34:06 valent: 14:34:06.9816     valent-lan-channel-service:  WARNING: valent_lan_channel_service_socket_send(): failed to identify: Error sending message: Cannot assign requested address
14:34:05 valent: 14:34:05.9686                       GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation portal (GNetworkMonitorPortal) for ‘gio-network-monitor’
14:34:05 valent: 14:34:05.7960                       GLib-GIO:    DEBUG: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
14:34:05 valent: 14:34:05.7710                       GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation gnutls (GTlsBackendGnutls) for ‘gio-tls-backend’
14:34:05 valent: 14:34:05.7693                       GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation keyfile (GKeyfileSettingsBackend) for ‘gsettings-backend’
14:34:05 valent: 14:34:05.7129                       GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
14:34:05 valent: 14:34:05.6066                       GLib-GIO:    DEBUG: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
14:34:05 systemd: Started app-flatpak-ca.andyholmes.Valent-1137.scope.
```

</details>

<details>

<summary>Debug info:</summary>

```
{
  "application" : {
    "id" : "ca.andyholmes.Valent",
    "version" : "1.0.0.alpha",
    "commit" : "508b7ca"
  },
  "runtime" : {
    "os" : "GNOME 45 (Flatpak runtime)",
    "desktop" : "GNOME",
    "session" : "wayland",
    "environment" : "flatpak"
  },
  "plugins" : {
    "xdp" : true,
    "telephony" : true,
    "systemvolume" : true,
    "sms" : true,
    "share" : true,
    "sftp" : true,
    "runcommand" : true,
    "pulseaudio" : true,
    "presenter" : true,
    "ping" : true,
    "notification" : true,
    "mpris" : true,
    "mousepad" : true,
    "lock" : true,
    "libvalent-ui" : true,
    "libvalent-device" : true,
    "lan" : true,
    "gtk" : true,
    "gnome" : true,
    "findmyphone" : true,
    "fdo" : true,
    "eds" : true,
    "contacts" : true,
    "connectivity_report" : true,
    "clipboard" : true,
    "battery" : true
  }
}

```

</details>

## postmarketOS phone
I'm currently figuring out where to find the logs on postmarketOS to share them here as well.

**Comment by andyholmes** [comment]
It looks like there is something funky happening with Mutter's D-Bus service on your systems. Are you running any patches or anything different about it from other Arch systems?

**Comment by ghassan0** [comment]
I don't have any Gnome/Mutter patches or made any changes regarding Mutter. As for gnome changes, it's mostly just switching to flatpaks from system packages for most applications, as well as install some extensions. [This](https://github.com/ghassan0/dotfiles/tree/main/gnome) basically documents any Gnome changes I made.

How do I find Valent's logs in postmarketOS? Running `logread | grep valent` didn't show anything.

**Comment by andyholmes** [comment]
> [This](https://github.com/ghassan0/dotfiles/tree/main/gnome) basically documents any Gnome changes I made.

I don't see anything there offhand that should interfere. I'm not sure why your Arch installs don't have the clipboard interface, but without it there's not much we can do on Wayland. Applications generally don't have access to the clipboard when running in the background, so the fallback `GdkClipboard` is not very useful.

> How do I find Valent's logs in postmarketOS? Running logread | grep valent didn't show anything.

Valent doesn't have it's own logging infrastructure, so if you're not running a logging service it's probably just not being recorded. If you are, it'll probably be in /var/log somewhere, but you'll have to check the docs.
